.github/workflows/deploy.yaml

```diff
@@ -29,45 +30,136 @@ jobs:
+        run: printf 'R_VERSION=%s\n' "$(jq --raw-output .R.Version ${{ github.event.repository.name }}/${{ matrix.directory }}/renv.lock)" >> $GITHUB_ENV
+
+      - name: Check if cypress test exists
+        id: find-cypress
         run: |
-          cat > /usr/local/lib/R/etc/Rprofile.site <<RPROFILE
-           options(
-            repos = c(
-              Pharmaverse = "https://pharmaverse.r-universe.dev",
-              CRAN = "https://cloud.r-project.org",
-              BioC = BiocManager::repositories()
-            ),
-            download.file.method = 'libcurl',
-            HTTPUserAgent = sprintf(
-              "R/%s R (%s)",
-              getRversion(),
-              paste(getRversion(), R.version["platform"], R.version["arch"], R.version["os"])
-            )
-          )
-          RPROFILE
+          if [ -d ${{ github.event.repository.name }}/${{ matrix.directory }}/tests/cypress ]; then
+            echo "has-cypress-tests=true" >> $GITHUB_OUTPUT
+          else
+            echo "Skipping frontend tests because cypress directory does not exist."
+            echo "has-cypress-tests=false" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Setup system dependencies for cypress and python app
+        if: steps.find-cypress.outputs.has-cypress-tests == 'true'
+        run: >
+          apt-get update && apt-get install --yes
+          libgtk2.0-0 libgbm-dev libnotify-dev libgconf-2-4 xvfb python3.10-venv
 
-      - name: Deploy apps 🖨 ${{ github.event.repository.name }} 🎨
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        if: steps.find-cypress.outputs.has-cypress-tests == 'true'
+        with:
+          node-version: 16
+
+      - name: Setup npm packages
+        if: steps.find-cypress.outputs.has-cypress-tests == 'true'
+        shell: bash
+        working-directory: ${{ github.event.repository.name }}/${{ matrix.directory }}/js
         run: |
-          Sys.setenv(GITHUB_PAT="${{ secrets.GITHUB_TOKEN }}")
-          devtools::install_local(
-            ".",
-            upgrade = "never",
-            dependencies = TRUE
+          npm i
+
+      - name: Restore renv from cache
+        uses: actions/cache@v2
+        env:
+          CACHE_KEY: renv-${{ runner.arch }}-${{ runner.os }}-${{ env.R_VERSION }}-${{ matrix.directory }}-${{ env.APP_SUFFIX }}
+        with:
+          path: ${{ github.event.repository.name }}/${{ matrix.directory }}/renv/library
+          key: ${{ env.CACHE_KEY }}-${{ hashFiles(format('{0}/{1}/renv.lock', github.event.repository.name, matrix.directory)) }}
+          restore-keys: ${{ env.CACHE_KEY }}-
+
+      - name: Install app dependencies
+        shell: Rscript {0}
+        working-directory: ${{ github.event.repository.name }}/${{ matrix.directory }}
+        run: |
+          options(renv.config.cache.symlinks = FALSE)
+          renv::restore(clean = TRUE)
+          install.packages(renv::dependencies()$Package)
+          renv::snapshot(update = TRUE)
```

Comment:
I feel that somewhere there we should add `setOptions(repos = ...)` before calling update. But I spotted that you have r-universe in the last auto-commit. So maybe there is something that I am missing.

Actutally, to me the sequnce should be as follows:
- `options(renv.(...))`
- `setOption()`
- `update()`
- `restore()`
- (and no install as restore should install the updated)
- auto-commit only on stable run

Reply:
The `Overwrite the global Rprofile ✍️` step adds the repo options to the Rprofile so we don't have to do it in every R shell. And I agree with the rest of the sequence. There is another possible sequence where we do not restore and built the `renv.lock` file from scratch, this logic will also make sure that we will update the renv package version (as recently the package was updated to version 1.0.0 which would have been missed in the previous sequence)

Reply:
Nevermind, I see that even the `renv` package along with the `renv/activate.R` gets updated after using `renv::update()`